cartridge/failover.lua

```diff
@@ -192,6 +200,49 @@ local function apply_config(mod)
     )
 end
 
+local function reconfigure_all()
+    local confapplier = require('cartridge.confapplier')
+    local all_roles = require('cartridge.roles').get_all_roles()
+
+    -- WARNING: implicit yield
+    -- The event may arrive while two-phase commit is in progress.
+    -- We should wait for the appropriate state.
+    local state = confapplier.wish_state('RolesConfigured', math.huge)
+    fiber.testcancel()
+
+    if state ~= 'RolesConfigured' then
+        log.info('Skipping failover step - state is %s', state)
+        return
+    end
+
+    fiber.self().storage.is_busy = true
+    confapplier.set_state('ConfiguringRoles')
+
+    local ok, err = FailoverError:pcall(function()
+        box.cfg({
+            read_only = not vars.cache.is_rw,
+        })
+
+        for _, role_name in ipairs(all_roles) do
+            local mod = service_registry.get(role_name)
+            local _, err = apply_config(mod)
+            if err then
+                log.error('Role %q failover failed', mod.role_name)
+                log.error('%s', err)
+            end
+        end
+
+        return true
+    end)
+
+    if ok then
+        log.info('Failover step finished')
+    else
+        log.warn('Failover step failed: %s', err)
+    end
+    confapplier.set_state('RolesConfigured')
```

Comment:
Seems we should turn off `is_busy` mode here

Reply:
It's not necessary. When the fiber is dead we can't access its storage anyway. So I check `fiber.status() == 'dead'` instead